pymatgen/core/structure.py

```diff
@@ -644,31 +644,40 @@ def extract_cluster(self, target_sites: list[Site], **kwargs) -> list[Site]:
             others = new_others
         return cluster
 
-    def _calculate(
-        self, struct: Structure | Molecule, calculator: str | Calculator, verbose: bool = False
-    ) -> Structure | Molecule:
+    def _calculate(self, calculator: str | Calculator, verbose: bool = False) -> Structure | Molecule:
         """
         Performs an ASE calculation.
 
         Args:
-            struct: Structure or Molecule to run ASE calculation on.
-            calculator: An ASE Calculator or a string from the following options: "m3gnet",
-                "gfn2-xtb".
+            calculator (str | Calculator): An ASE Calculator or a string from the following case-insensitive
+                options: "m3gnet", "gfn2-xtb", "chgnet".
             verbose (bool): whether to print stdout. Defaults to False.
+                Has no effect when calculator=='chgnet'.
 
         Returns:
-            Structure: Structure or Molelcule following ASE calculation.
+            Structure | Molecule: Structure or Molecule with new calc attribute containing
+                result of ASE calculation.
         """
+        if isinstance(calculator, str) and calculator.lower() == "chgnet":
+            from chgnet.model import CHGNet
+
+            chgnet = CHGNet.load()
+            preds = chgnet.predict_structure(self)
+
+            out = self.copy()  # type: ignore[attr-defined]
```

Comment:
Could you use a more descriptive variable name than `preds` and `out`?

Reply:
Got rid of this whole section entirely.